app/services/user/bulk_destroy.rb

```diff
@@ -17,64 +19,30 @@ class User::BulkDestroy
   private
 
   def fetch_users(params:)
-    ids = params.user_ids
-    # this order cluase ensures we retrieve the users in the same order as the
+    # this order clause ensures we retrieve the users in the same order as the
     # IDs in the param. we do this to ensure the users are deleted in the same
     # order as they're selected in the UI
-    User.where(id: ids).order(DB.sql_fragment("array_position(ARRAY[?], users.id)", ids))
+    User
+      .where(id: params.user_ids)
+      .order(DB.sql_fragment("array_position(ARRAY[?], users.id)", params.user_ids))
+      .to_a
   end
 
   def can_delete_users(guardian:, users:)
-    users.all? { |u| guardian.can_delete_user?(u) }
+    users.all? { guardian.can_delete_user?(_1) }
   end
 
   def delete(users:, guardian:, params:)
-    actor_ip = guardian.user.ip_address
     users
-      .find_each
+      .each
```

Comment:
As we can’t have more than 100 users at once, `#find_each` wasn’t that useful since the default size for its batch is 1000. Moreover, we were calling `users.size` in the loop which was issuing a count statement each time since `users` was an AR relation object and not an array.